product_variant_default_code/models/product.py

```diff
@@ -100,12 +104,34 @@ class ProductTemplate(models.Model):
         "attribute name",
     )
 
+    @api.depends("code_prefix", "attribute_line_ids")
+    def _compute_reference_mask(self):
+        automask = bool(
+            not self.user_has_groups(
+                "product_variant_default_code.group_product_default_code_manual_mask"
+            )
+        )
+        for rec in self:
+            if automask or rec.reference_mask == "":
+                rec.reference_mask = rec._get_default_mask()
+            rec._apply_reference_mask()
```

Comment:
instead of calling manually the _apply_reference_mask.
It will be better to add a computed field on the default_code of the product_product.

somethink like
```
@api.depends(
    "product_tmpl_id.code_prefix",
    "product_template_attribute_value_ids.product_attribute_value_id.code",
     "product_template_attribute_value_ids.attribute_id.code",
)
def _compute_default_code(self):
   ...

```

Reply:
@sebastienbeau, thanks for your suggestions, I apply them.